blog: Nuxt 2 EOL (#1599)
nuxt/nuxt.com: content/7.blog/4.nuxt2-eol.md

---
title: Nuxt 2 End-of-Life (EOL)
description: "Nuxt 2 will reach End of Life (EOL) on June 30th, 2024. We've partnered with HeroDevs on offering Never-Ending Support (NES)."
image: /assets/blog/introducing-nuxt-3-beta.png
date: 2024-06-14
authors:

>   - name: Sébastien Chopin
>     avatarUrl: https://github.com/atinux.png
>     link: https://twitter.com/atinux

Author: ```suggestion
  - name: Sébastien Chopin
    avatarUrl: https://github.com/atinux.png
    link: https://x.com/atinux
  - name: Daniel Roe
    avatarUrl: https://github.com/danielroe.png
    link: https://x.com/danielcroe
```

::note
Vue 2 reached its end-of-life date on December 31st, 2023.
::
    
## What’s Next?

Nuxt 3 has been the default version of Nuxt since November 16th, 2022.

On top of using Vue 3 and the composition API, it is shipped with features and innovations:
- Universal & Hybrid Rendering: Benefits of both SPA and SSR, with fine-grained control over route rendering
- Support for serverless environments (AWS Lambda, Deno, Cloudflare Workers) with minimal cold-start.
- First-Class TypeScript Support: Full typing across all components and configurations.
- Vite integration for a faster developer experience
- Server & API routes with end-to-end typing powered by Nitro
- Auto import of composables & utils
- Layers feature for domain driven development

When and if you can, consider migrating to Nuxt 3 to take advantage of these powerful features.

::read-more{to="/docs/getting-started/upgrade#nuxt-2-vs-nuxt-3"}
See a full comparison table between Nuxt 2 and Nuxt 3, as well as detailed upgrade guides to Nuxt 3 and Nuxt Bridge (as a part of upgrading to Nuxt 3).
::

> 

Reviewer: ```suggestion

::read-more{to="/enterprise/support"}
If you need support, including upgrading to Nuxt 3, NuxtLabs provides professional support and consultancy in a wide range of areas.
::

```

---
title: Nuxt 2 End-of-Life (EOL)
description: "Nuxt 2 will reach End of Life (EOL) on June 30th, 2024. We've partnered with HeroDevs on offering Never-Ending Support (NES)."
image: /assets/blog/introducing-nuxt-3-beta.png
date: 2024-06-14
authors:
  - name: Sébastien Chopin
    avatarUrl: https://github.com/atinux.png
    link: https://x.com/atinux
  - name: Daniel Roe
    avatarUrl: https://github.com/danielroe.png
    link: https://x.com/danielcroe

> category: Release

Author: ```suggestion
category: Announcement
```

date: 2024-06-14
authors:
  - name: Sébastien Chopin
    avatarUrl: https://github.com/atinux.png
    link: https://twitter.com/atinux
category: Release
---

Released in 2018, Nuxt 2.0 marked a major milestone, establishing it as a mainstream framework. Over the past six years, many developers adopted Nuxt 2, leading to the creation of numerous impressive projects across the web.

However, looking forward, maintaining Nuxt 2 is no longer sustainable. With the recent end-of-life of Vue 2 and the maturity of Nuxt 3 and its ecosystem, it's time for our team to concentrate our efforts on the latest major version and upcoming versions.

::tip{to="https://www.herodevs.com/support/nuxt-nes?utm_source=nuxt&utm_medium=nuxt-eol-article" target="_blank"}
Jump over HeroDevs' Nuxt Never-Ending Support (NES)
::

## What happens on June 30th, 2024?

After this date, Nuxt 2 will continue to be available on the NPM package manager, but will no longer receive updates, such as security and browser compatibility fixes. In other words, your applications will continue to work, but you may get deprecation warnings from your package manager reminding you that Nuxt 2 is no longer a supported version.

> <!--
> On June 30th, the following packages will be marked as deprecated on NPM:
> - `nuxt@v2`
> -->
> 

Author: ```suggestion
```

::note
Vue 2 reached its end-of-life date on December 31st, 2023.
::
    
## What’s Next?

Nuxt 3 has been the default version of Nuxt since November 16th, 2022.

On top of using Vue 3 and the composition API, it is shipped with features and innovations:
- Universal & Hybrid Rendering: Benefits of both SPA and SSR, with fine-grained control over route rendering
- Support for serverless environments (AWS Lambda, Deno, Cloudflare Workers) with minimal cold-start.
- First-Class TypeScript Support: Full typing across all components and configurations.
- Vite integration for a faster developer experience
- Server & API routes with end-to-end typing powered by Nitro
- Auto import of composables & utils
- Layers feature for domain driven development

When and if you can, consider migrating to Nuxt 3 to take advantage of these powerful features.

::read-more{to="/docs/getting-started/upgrade#nuxt-2-vs-nuxt-3"}

> See a full comparison table between Nuxt 2 and Nuxt 3.

Reviewer: ```suggestion
See a full comparison table between Nuxt 2 and Nuxt 3, as well as detailed upgrade guides to Nuxt 3 and Nuxt Bridge (as a part of upgrading to Nuxt 3).
```

::

## Still on Nuxt 2? Here Are Your Options.

Recognizing the various situations that arise during transitions, we are fully aware that users may need other options until they can migrate, or maybe migration simply isn't a feasible path. Here are some other options to consider:

### Update to the Nuxt 2 latest release

> The to-be-released EOL version (2.17.4) will be the final release of Nuxt 2, planned for June 2024. This patch release includes a few final fixes, improving type alignment with Nuxt 3. 
> 
> ::note
> We strongly encourage you to update to 2.17.4 once it's out. This will be the starting point for extended support mentioned below.
> ::

Reviewer: ```suggestion
The to-be-released EOL version (2.18.0) will be the final release of Nuxt 2, planned for June 2024. This release will include the last few final fixes.

::note
We strongly encourage you to update to 2.18.0 once it's out. This will be the starting point for extended support mentioned below.
::
```
